python/plugins/processing/tools/vector.py

```diff
@@ -104,6 +108,27 @@ def __init__(self, layer, request):
                 self.selection = True
             else:
                 self.iter = layer.getFeatures(request)
+            
+            invalidFeaturesMethod = ProcessingConfig.getSetting(ProcessingConfig.FILTER_INVALID_GEOMETRIES)
+            
+            def filterFeature(f, ignoreInvalid):
+                geom = f.geometry()
+                if geom is None:
+                    ProcessingLog.addToLog(ProcessingLog.LOG_INFO,
+                                            self.tr('Feature with NULL geometry found.'))
+                elif not geom.isGeosValid():
+                    ProcessingLog.addToLog(ProcessingLog.LOG_ERROR,
+                                            self.tr('GEOS geoprocessing error: One or more input features have invalid geometry.'))
+                    if ignoreInvalid:
+                        return False
+                    else:
+                        raise GeoAlgorithmExecutionException(self.tr('Features with invalid geometries found. Please fix these errors or specify the "Ignore invalid input features" flag'))
```

Comment:
That would be the nicest solution of course, if that's going to happen my comment is pointless.